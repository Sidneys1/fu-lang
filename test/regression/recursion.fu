foo: u8(y: u8) = {
  if (y == 1) return y;
  return y + foo(y - 1);
};

main: u8() => foo(4); // 4 + (3 + (2 + 1)) = 10
