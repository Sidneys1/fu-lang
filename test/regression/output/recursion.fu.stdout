----------- test\regression\recursion.fu -----------
[2m   3 |   return y + foo([0;2my - 1[2m);[0m[0;0m <-- Checks for infix operator '-' are not implemented! [0;2m(test\regression\recursion.fu:3:18-22)
[0m[m[0;2m-----+[0m
[2m   3 |   return [0;2my + foo(y - 1);[2m[0m[0;0m <-- Checks for infix operator '+' are not implemented! [0;2m(test\regression\recursion.fu:3:10-24)
[0m[m